docs/api-docs/apps/tutorials/sample-app-next/04-step-2-connect-to-bc.md

```diff
@@ -0,0 +1,296 @@
+
+3. Open the `auth.ts` file and add the `BigCommerce` import at the top of the file.
+
+```js
+import * as BigCommerce from 'node-bigcommerce';
+```
+
+4. Create a BigCommerce instance required as part of the authorization step when first installing the application.
+
+```js
+// Create BigCommerce instance
+// https://github.com/getconversio/node-bigcommerce
+const bigcommerce = new BigCommerce({
+    logLevel: 'info',
+    clientId: process.env.CLIENT_ID,
+    secret: process.env.CLIENT_SECRET,
+    callback: process.env.AUTH_CALLBACK,
+    responseType: 'json',
+    headers: { 'Accept-Encoding': '*' },
+    apiVersion: 'v3'
+});
+
+const bigcommerceSigned = new BigCommerce({
+    secret: process.env.CLIENT_SECRET,
+    responseType: 'json'
+});
+
+interface QueryParams {
+    [key: string]: string;
+}
+```
+
+The `bigcommerceSigned` function will be called when loading or uninstalling the application.
+
+5. Export the `getBCAuth` function.
+
+```js
+export function getBCAuth(query: QueryParams) {
+    return bigcommerce.authorize(query);
+}
+```
+
+The `authorize` method will retrieve your permanent access token and return it in the `data` object.
+
+6. Export the `getBCVerify` function.
+
+```js
+export function getBCVerify({ signed_payload }: QueryParams) {
+    return bigcommerceSigned.verify(signed_payload);
+}
+```
+
+The `verify` method employs the `signed_payload` query parameter to authenticate requests. You will use the `verify` method for both `/load` and `/uninstall` API endpoints. To learn more about the OAuth flow, see [Single-Click App OAuth Flow](https://developer.bigcommerce.com/api-docs/apps/guide/auth).
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-2-connect-app-to-bc/lib/auth.ts)
+
+## Add API endpoints
+
+Next.js maps all APIs that are part of the Next.js package to the `/api/*` route. You can take advantage of it by creating a sub-directory within the `pages` folder called `api`. This will signal Next.js to treat files within `/apges/api` as API endpoints. Instead of looking for React components, Next.js will look for and handle API routes. Next.js will also automatically handle routing for files placed within `/pages/api`. To learn more about Next.JS API routes, see [API Routes](https://nextjs.org/docs/api-routes/introduction). 
+
+1. In the `pages` folder, create an `api` folder.
+
+2. Open the api folder and add the following files: `auth.ts`, `load.ts`, and `uninstall.ts`.
+
+3. Open the `auth.ts` file and add the logic to authorize the app.
+
+```js
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getBCAuth } from '../../lib/auth';
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+    try {
+        await getBCAuth(req.query);
+
+        res.redirect(302, '/');
+    } catch (error) {
+        const { message, response } = error;
+        res.status(response?.status || 500).json(message);
```

Comment:
this is my fault: we recently corrected this line... should be `.json({ message });`, with the message wrapped in an object.  If message isn't wrapped in an object, `.json` won't properly convert